resin/settings.py

```diff
@@ -35,15 +36,20 @@ class Settings(object):
                                             'cache')
 
     def __init__(self):
-        if Path.isdir(self._setting['data_directory']):
+        try:
+            self.__read_settings()
+        except:
+            # move old settings file if it exists.
             if Path.isfile(Path.join(self._setting['data_directory'],
                                      self.CONFIG_FILENAME)):
```

Comment:
The config's path is used several times in this method, so you could assign it once to a local variable and use that. It'll make the code easier to read. ``config_path = Path.join(self._setting['data_directory'], self.CONFIG_FILENAME)``